### DISCUSSION TITLE
feeding the py5 "integrations radar" - graphs and geospacial data

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#484] [Ideas]
Now that py5 has wonderful [shapely integrations](https://py5coding.org/integrations/shapely.html), I wonder if the next step could be something about, graphs ([networkx](https://networkx.org/)), geodataframes ([geopandas](https://geopandas.org/en/stable/getting_started/introduction.html)), and things in between, like [osmnx](https://osmnx.readthedocs.io/en/stable/index.html) that taps into OpenStreetMaps data to build a networkx graph object and can produce geodataframes which use shapely objects in the geometry column.

I'm already playing with this, slow pace, trying to figure ways of combining osmnx with the geometry of geodataframe layers provided by Marcelo Prates' great [prettymaps](https://github.com/marceloprates/prettymaps)... I might post more at the "show and tell" category if I get confident enough.

**Comment by hx2A** [comment]
I've often thought about networkx and py5, and how they could work together. networkx is an excellent library. I can't think of how exactly it would translate into shapes though...I'm eager to see what you come up with! geopandas could be a great fit. It uses shapely geometries as a datatype, on top of Pandas. I don't think it would be that difficult to include something there. osmnx is an interesting choice also but I think the user would have a lot of opinions about how the data was translated into shapes.

In any case, py5's shapely integrations actually started with you, so I'll follow closely with what you do and I know that will inspire ideas for how it can work well with py5.
